File: js/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let i = 0;
    let hash = {};
    let ans = 0;
    
    for (let i = 0; i < nums.length; i++) {
        hash[nums[i]] = i;
    }
    
    for (let i = 0; i < nums.length; i++) {
        let temp = target - nums[i];
        if (hash[temp] && hash[temp] != i) {
            return [i, hash[temp]];
        }
    }
    
    return [-1, -1];
};

File: java/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
         HashMap <Integer, Integer> hash = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            hash.put(nums[i], i);
        }
        
        for (int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            if (hash.containsKey(temp) && i != hash.get(temp)) {
                return new int[]{i, hash.get(temp)};
            }
        }
        
        return new int[]{};
    }
}